src/main/java/com/puppycrawl/tools/checkstyle/checks/design/FinalClassCheck.java

```diff
@@ -253,14 +253,27 @@ public void leaveToken(DetailAST ast) {
      * @return true if given class should be declared as final otherwise false
      */
     private static boolean shouldBeDeclaredAsFinal(ClassDesc desc) {
-        return desc.isWithPrivateCtor()
+        return (desc.isWithPrivateCtor() || isClassPrivate(desc))
```

Comment:
in this push i have added ` (!desc.isWithPrivateCtor() && isClassPrivate(desc))` && !descWithPrivateCTor is also their this leads to coverage and pitest error so only isClassPrivate is their in previous push

Reply:
> please make Input

did you make it ?

Reply:
@romani I have tried to make a method to check the default constructor like 
```
    private static boolean isClassPrivate(DetailAST desc) {
        return desc.getFirstChild().findFirstToken(TokenTypes.LITERAL_PRIVATE) != null;
    }

    private static boolean isDefaultConstructor(DetailAST desc) {
        if(isClassPrivate(desc)) {
            return desc.getLastChild().findFirstToken(TokenTypes.CTOR_DEF) == null;
        }
        return false;
    }
```

it leads to code coverage error i have tried to implement it in various type and add test case much as possible but it is been not figured out to make the method like this. The code coverage is because of return statement 
it will check if their is private constructor it will check if their is non private constructor 

we have to check for default constructor for only private class so this requirement is first and second to check if ctor is null 
but as i said in return statement it is been already checking if their is constructor so currently i have only put the condition to check private class

Reply:
the one test trhough which we can implement the code i share but i need a confirmation from you I am not sure 


```
class Constructor {
    private class con {  
        con(){}  // constructor line --> 3
    }
}

```

what if their is constructor in private class without modifier is `default` or `private`

Reply:
@romani https://github.com/checkstyle/checkstyle/pull/12737#discussion_r1148101344 I think the constructor at line 3 is private and class should be declared as final please let me know if i am wrong at this point. 

using this test i think we can check for default constructor. without any issue

Reply:
@romani please give suggestion here https://github.com/checkstyle/checkstyle/pull/12737#discussion_r1148321868

Reply:
No modifier means package-private, It is not a private as it is visible to all in same package, let's not demand final on it.

Reply:
@romani please check I have updated the branch and implement method to check check if their is default constructor
no coverage issue is their